File: examples/functions/rest-api-methods/posts/[slug]/get.go
package main

import "encoding/json"

// GET /posts/:slug — slug merged into event from [slug] filename
func handler(event map[string]interface{}) interface{} {
	slug, _ := event["slug"].(string)

	body, _ := json.Marshal(map[string]interface{}{
		"slug": slug, "title": "Post: " + slug, "content": "Lorem ipsum...",
	})
	return map[string]interface{}{
		"status":  200,
		"headers": map[string]string{"Content-Type": "application/json"},
		"body":    string(body),
	}
}

File: examples/functions/rest-api-methods/posts/[slug]/get.rs
use serde_json::{json, Value};

// GET /posts/:slug — slug merged into event from [slug] filename
pub fn handler(event: Value) -> Value {
    let slug = event["slug"].as_str().unwrap_or("");

    json!({
        "status": 200,
        "headers": { "Content-Type": "application/json" },
        "body": serde_json::to_string(&json!({
            "slug": slug, "title": format!("Post: {}", slug), "content": "Lorem ipsum..."
        })).unwrap()
    })
}
